controllers/subscriptions.go

```diff
@@ -102,6 +114,23 @@ func CheckExistingSubscriptions(cli client.Client, desiredSubscription *operator
 	return desiredSubscription, nil
 }
 
+func isProviderServer(cli client.Client) (bool, error) {
+
+	storageclusters := &ocsv1.StorageClusterList{}
+	err := cli.List(context.TODO(), storageclusters)
+	if err != nil {
+		return false, err
+	}
+
+	for _, storagecluster := range storageclusters.Items {
+		if storagecluster.Spec.AllowRemoteStorageConsumers {
+			return true, nil
+		}
+	}
+
+	return false, nil
```

Comment:
sry, this seems a lot of work to find the fields values, not mandatory and here's what I'm thinking, `cli.List` will try to list storageclusters in all namespaces and `storagecluster` in for loop is making a copy of every storagecluster in the Items.

It might be good to limit the result to operator ns and taken an index into Items? again, only a suggestion.

Reply:
I would say let it be. We may extend the func later on.